main.go

```diff
@@ -11,12 +11,12 @@ import (
 type nodeType int
 
 const (
-	join nodeType = iota
-	scan
+	joinNode nodeType = iota
+	scanNode
```

Comment:
Now that there is an `operator` interface, I think the uses of `nodeType` can be replaced with type switches or something better. I'll take a crack at doing this when this PR is merged.